mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs

```diff
@@ -97,9 +97,39 @@ public MethodBase Resolve ()
 			else
 				sig_cand = type.GetMethod (methodName, BindingFlags.Public|BindingFlags.NonPublic|BindingFlags.Instance, null, param_types, null);
 
+			if (sig_cand != null && generic_arg_names != null && !sig_cand.IsGenericMethodDefinition)
```

Comment:
Why is this affected by dedup which should have no visible effect on execution ?

Reply:
The mkbundle PR moves the test dlls out of the current directory into a subdirectory of the profile dir in lib. When the test is run out of the mcs/class subdir, the dll is then not in the current working directory. On windows, assemblies in the current working directory end up loaded in running domains, so the cross-appdomain call code uses the fast path that doesn't use the CADMessage remoting code. By moving the .dll, we made that dead code run on Windows. This dead code had a bug when the method is inflated, and one of the parameters ends up in the argument. That argument gets marshaled by type name, and the AssemblyQualifiedName of that MVAR returns null, putting a null in that array. When we later try to fetch the type and pass a null, it throws.